packages/cli/src/commands/wf/sops-set-profile/command.ts

```
@@ -0,0 +1,93 @@
+import { PanfactumCommand } from '@/util/command/panfactumCommand';
+
+export class SopsSetProfileCommand extends PanfactumCommand {
+  static override paths = [['wf', 'sops-set-profile']];
+
+  static override usage = Command.Usage({
+    description: 'Update AWS profile used for KMS access in SOPS-encrypted YAML files',
+    details: `Updates the AWS profile used to access KMS in all sops-encrypted YAML files in the indicated directory tree.
+This can be used in CI pipelines to simplify access to encrypted files that would otherwise require many AWS profiles to be configured.`,
+    examples: [
+      ['Update profile in current directory', 'pf sops set-profile . development'],
+      ['Update profile in specific directory', 'pf sops set-profile /path/to/dir production']
+    ]
+  });
+
+  directory = Option.String({ required: true });
+  profile = Option.String({ required: true });
+
+  async execute() {
+    const updateSopsFile = async (filePath: string): Promise<boolean> => {
+      try {
+        const content = await fs.readFile(filePath, 'utf8');
+        
+        const sopsFileSchema = z.object({
+          sops: z.object({
+            kms: z.array(z.record(z.string())).optional()
+          }).optional()
+        }).passthrough(); // Allow other fields in the YAML
+        
+        const parsedYaml = yaml.parse(content);
+        const data = sopsFileSchema.parse(parsedYaml);
+        
+        if (data?.sops?.kms && Array.isArray(data.sops.kms)) {
+          let updated = false;
+          for (const kms of data.sops.kms) {
+            if ('aws_profile' in kms) {
+              kms['aws_profile'] = this.profile;
+              updated = true;
+            }
+          }
+          
+          if (updated) {
+            const newContent = yaml.stringify(data);
+            await fs.writeFile(filePath, newContent);
+            return true;
+          }
+        }
+      } catch {
+        // Not a valid YAML or doesn't have SOPS metadata
```

Never have catch alls that swallow every error